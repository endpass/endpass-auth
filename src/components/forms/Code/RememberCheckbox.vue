<template>
  <v-checkbox
    v-bind="$attrs"
    type="checkbox"
    v-on="$listeners"
  >
    {{ $t('components.rememberCheckbox.text') }}
  </v-checkbox>
</template>

<script>
import VCheckbox from '@endpass/ui/kit/VCheckbox';

export default {
  name: 'RememberCheckbox',

  components: {
    VCheckbox,
  },

  model: {
    prop: 'value',
    event: 'change',
  },
};
</script>
