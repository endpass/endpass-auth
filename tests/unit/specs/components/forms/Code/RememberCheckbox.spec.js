import { shallowMount, createLocalVue } from '@vue/test-utils';
import RememberCheckbox from '@/components/forms/Code/RememberCheckbox';

import setupI18n from '@/locales/i18nSetup';

const localVue = createLocalVue();
const i18n = setupI18n(localVue);

describe('RememberCheckbox', () => {
  let wrapper;

  const createWrapper = options => {
    return shallowMount(RememberCheckbox, {
      localVue,
      i18n,
      sync: false,
      ...options,
    });
  };

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = createWrapper();
  });

  describe('render', () => {
    it('should correctly render component', async () => {
      expect.assertions(2);

      expect(wrapper.name()).toBe('RememberCheckbox');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
