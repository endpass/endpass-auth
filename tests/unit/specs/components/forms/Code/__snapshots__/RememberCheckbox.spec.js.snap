// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RememberCheckbox render should correctly render component 1`] = `
<v-checkbox-stub name="checkbox" truevalue="true" type="checkbox">
  Remember me on this device
</v-checkbox-stub>
`;
